From 7f7c41cfdf6a9db2c7ec336c76a70219d008ac29 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sat, 22 Aug 2026 00:59:06 +0000
Subject: [PATCH 04/36] Imap, tests: convert new-style connects to SIGNAL/SLOT
 syntax

Qt4 has no support for the &Class::member connect syntax; converted
every connect()/disconnect() in src/Imap and tests to the equivalent
string-based form, verifying each signature against its declaring
class (adding missing slots: declarations where needed).
---
 src/Imap/Model/CombinedCache.cpp              |  4 +-
 src/Imap/Model/FullMessageCombiner.cpp        |  4 +-
 src/Imap/Model/FullMessageCombiner.h          |  1 -
 src/Imap/Model/ImapAccess.cpp                 | 18 +++---
 src/Imap/Model/MailboxFinder.cpp              |  4 +-
 src/Imap/Model/Model.cpp                      |  4 +-
 src/Imap/Model/Model.h                        |  1 +
 src/Imap/Model/ModelWatcher.cpp               | 38 ++++++------
 src/Imap/Model/NetworkWatcher.cpp             |  8 +--
 src/Imap/Model/OneMessageModel.cpp            |  8 +--
 src/Imap/Model/SQLCache.cpp                   |  4 +-
 src/Imap/Model/SubtreeModel.cpp               | 26 ++++----
 src/Imap/Model/TaskFactory.cpp                | 10 ++--
 src/Imap/Model/ThreadingMsgListModel.cpp      | 60 +++++++++----------
 src/Imap/Network/FileDownloadManager.cpp      | 15 +++--
 src/Imap/Network/MsgPartNetworkReply.cpp      |  2 +-
 .../Network/QQuickNetworkReplyWrapper.cpp     |  2 +-
 src/Imap/Parser/Parser.cpp                    |  8 +--
 src/Imap/Tasks/FetchMsgPartTask.cpp           |  2 +-
 src/Imap/Tasks/IdleLauncher.cpp               |  4 +-
 src/Imap/Tasks/ImapTask.cpp                   |  4 +-
 src/Imap/Tasks/KeepMailboxOpenTask.cpp        | 20 +++----
 .../Tasks/ObtainSynchronizedMailboxTask.cpp   |  4 +-
 src/Imap/Tasks/OpenConnectionTask.cpp         |  8 +--
 src/Imap/Tasks/UnSelectTask.cpp               |  2 +-
 tests/Composer/test_Html_formatting.cpp       |  2 +-
 .../Imap/test_Imap_SelectedMailboxUpdates.cpp |  2 +-
 .../test_Imap_Tasks_ListChildMailboxes.cpp    | 28 +++++----
 .../Imap/test_Imap_Tasks_ListChildMailboxes.h | 10 ++++
 tests/Imap/test_Imap_Tasks_OpenConnection.cpp |  4 +-
 tests/Imap/test_Imap_Threading.cpp            |  4 +-
 tests/Utils/LibMailboxSync.cpp                |  4 +-
 32 files changed, 166 insertions(+), 149 deletions(-)

diff --git a/src/Imap/Model/CombinedCache.cpp b/src/Imap/Model/CombinedCache.cpp
index 06096a10..8287f4a0 100644
--- a/src/Imap/Model/CombinedCache.cpp
+++ b/src/Imap/Model/CombinedCache.cpp
@@ -33,9 +33,9 @@ CombinedCache::CombinedCache(QObject *parent, const QString &name, const QString
     AbstractCache(parent), name(name), cacheDir(cacheDir)
 {
     sqlCache = new SQLCache(this);
-    connect(sqlCache, &AbstractCache::error, this, &AbstractCache::error);
+    connect(sqlCache, SIGNAL(error(QString)), this, SIGNAL(error(QString)));
     diskPartCache = new DiskPartCache(this, cacheDir);
-    connect(diskPartCache, &DiskPartCache::error, this, &AbstractCache::error);
+    connect(diskPartCache, SIGNAL(error(QString)), this, SIGNAL(error(QString)));
 }
 
 CombinedCache::~CombinedCache()
diff --git a/src/Imap/Model/FullMessageCombiner.cpp b/src/Imap/Model/FullMessageCombiner.cpp
index cf44ba0b..67b38aa5 100644
--- a/src/Imap/Model/FullMessageCombiner.cpp
+++ b/src/Imap/Model/FullMessageCombiner.cpp
@@ -39,7 +39,7 @@ FullMessageCombiner::FullMessageCombiner(const QModelIndex &messageIndex, QObjec
     Q_ASSERT(m_headerPartIndex.isValid());
     m_bodyPartIndex = m_messageIndex.child(0, Imap::Mailbox::TreeItem::OFFSET_TEXT);
     Q_ASSERT(m_bodyPartIndex.isValid());
-    m_dataChanged = connect(m_messageIndex.model(), &QAbstractItemModel::dataChanged, this, &FullMessageCombiner::slotDataChanged);
+    connect(m_messageIndex.model(), SIGNAL(dataChanged(QModelIndex,QModelIndex)), this, SLOT(slotDataChanged(QModelIndex,QModelIndex)));
 }
 
 QByteArray FullMessageCombiner::data() const
@@ -82,7 +82,7 @@ void FullMessageCombiner::slotDataChanged(const QModelIndex &left, const QModelI
 
     if (m_headerPartIndex.data(Imap::Mailbox::RoleIsFetched).toBool() && m_bodyPartIndex.data(Imap::Mailbox::RoleIsFetched).toBool()) {
         emit completed();
-        disconnect(m_dataChanged);
+        disconnect(m_messageIndex.model(), SIGNAL(dataChanged(QModelIndex,QModelIndex)), this, SLOT(slotDataChanged(QModelIndex,QModelIndex)));
     }
 
     bool headerOffline = m_headerPartIndex.data(Imap::Mailbox::RoleIsUnavailable).toBool();
diff --git a/src/Imap/Model/FullMessageCombiner.h b/src/Imap/Model/FullMessageCombiner.h
index 85a6c3df..2b7dfd31 100644
--- a/src/Imap/Model/FullMessageCombiner.h
+++ b/src/Imap/Model/FullMessageCombiner.h
@@ -64,7 +64,6 @@ private:
     QPersistentModelIndex m_bodyPartIndex;
     QPersistentModelIndex m_headerPartIndex;
     QPersistentModelIndex m_messageIndex;
-    QMetaObject::Connection m_dataChanged;
 };
 
 
diff --git a/src/Imap/Model/ImapAccess.cpp b/src/Imap/Model/ImapAccess.cpp
index f2c1adb7..36866425 100644
--- a/src/Imap/Model/ImapAccess.cpp
+++ b/src/Imap/Model/ImapAccess.cpp
@@ -240,7 +240,7 @@ void ImapAccess::doConnect()
     if (m_netWatcher) {
         // We're temporarily "disabling" this connection. Otherwise this "offline preference"
         // would get saved into the config file, which would be bad.
-        disconnect(m_netWatcher, &Mailbox::NetworkWatcher::desiredNetworkPolicyChanged, this, &ImapAccess::desiredNetworkPolicyChanged);
+        disconnect(m_netWatcher, SIGNAL(desiredNetworkPolicyChanged(Imap::Mailbox::NetworkPolicy)), this, SLOT(desiredNetworkPolicyChanged(Imap::Mailbox::NetworkPolicy)));
     }
 
     if (m_imapModel) {
@@ -326,7 +326,7 @@ void ImapAccess::doConnect()
         cache = new Imap::Mailbox::MemoryCache(this);
     } else {
         cache = new Imap::Mailbox::CombinedCache(this, QStringLiteral("trojita-imap-cache"), m_cacheDir);
-        connect(cache, &Mailbox::AbstractCache::error, this, &ImapAccess::onCacheError);
+        connect(cache, SIGNAL(error(QString)), this, SLOT(onCacheError(QString)));
         if (! static_cast<Imap::Mailbox::CombinedCache *>(cache)->open()) {
             // Error message was already shown by the cacheError() slot
             cache->deleteLater();
@@ -351,19 +351,19 @@ void ImapAccess::doConnect()
     m_imapModel->setProperty("trojita-imap-id-no-versions", !m_settings->value(Common::SettingsNames::interopRevealVersions, true).toBool());
     m_imapModel->setProperty("trojita-imap-idle-renewal", m_settings->value(Common::SettingsNames::imapIdleRenewal).toUInt() * 60 * 1000);
     m_imapModel->setNumberRefreshInterval(numberRefreshInterval());
-    connect(m_imapModel, &Mailbox::Model::alertReceived, this, &ImapAccess::alertReceived);
-    connect(m_imapModel, &Mailbox::Model::imapError, this, &ImapAccess::imapError);
-    connect(m_imapModel, &Mailbox::Model::networkError, this, &ImapAccess::networkError);
-    //connect(m_imapModel, &Mailbox::Model::logged, this, &ImapAccess::slotLogged);
-    connect(m_imapModel, &Mailbox::Model::needsSslDecision, this, &ImapAccess::slotSslErrors);
-    connect(m_imapModel, &Mailbox::Model::requireStartTlsInFuture, this, &ImapAccess::onRequireStartTlsInFuture);
+    connect(m_imapModel, SIGNAL(alertReceived(QString)), this, SLOT(alertReceived(QString)));
+    connect(m_imapModel, SIGNAL(imapError(QString)), this, SLOT(imapError(QString)));
+    connect(m_imapModel, SIGNAL(networkError(QString)), this, SLOT(networkError(QString)));
+    //connect(m_imapModel, SIGNAL(logged(uint,Common::LogMessage)), this, SLOT(slotLogged(uint,Common::LogMessage)));
+    connect(m_imapModel, SIGNAL(needsSslDecision(QList<QSslCertificate>,QList<QSslError>)), this, SLOT(slotSslErrors(QList<QSslCertificate>,QList<QSslError>)));
+    connect(m_imapModel, SIGNAL(requireStartTlsInFuture()), this, SLOT(onRequireStartTlsInFuture()));
 
     if (m_settings->value(Common::SettingsNames::imapNeedsNetwork, true).toBool()) {
         m_netWatcher = new Imap::Mailbox::SystemNetworkWatcher(this, m_imapModel);
     } else {
         m_netWatcher = new Imap::Mailbox::DummyNetworkWatcher(this, m_imapModel);
     }
-    connect(m_netWatcher, &Mailbox::NetworkWatcher::desiredNetworkPolicyChanged, this, &ImapAccess::desiredNetworkPolicyChanged);
+    connect(m_netWatcher, SIGNAL(desiredNetworkPolicyChanged(Imap::Mailbox::NetworkPolicy)), this, SLOT(desiredNetworkPolicyChanged(Imap::Mailbox::NetworkPolicy)));
     switch (preferredNetworkPolicy()) {
     case Imap::Mailbox::NETWORK_OFFLINE:
         QMetaObject::invokeMethod(m_netWatcher, "setNetworkOffline", Qt::QueuedConnection);
diff --git a/src/Imap/Model/MailboxFinder.cpp b/src/Imap/Model/MailboxFinder.cpp
index c6a627cc..b3fdf0d6 100644
--- a/src/Imap/Model/MailboxFinder.cpp
+++ b/src/Imap/Model/MailboxFinder.cpp
@@ -43,8 +43,8 @@ MailboxFinder::MailboxFinder(QObject *parent, QAbstractItemModel *model)
     Q_ASSERT(m_model);
     Q_ASSERT_X(!qobject_cast<Imap::Mailbox::Model *>(model),
                "MailboxFinder", "This requires a proxy model which exports a filtered view, not the original Model");
-    connect(m_model, &QAbstractItemModel::layoutChanged, this, &MailboxFinder::checkArrivals);
-    connect(m_model, &QAbstractItemModel::rowsInserted, this, &MailboxFinder::checkArrivals);
+    connect(m_model, SIGNAL(layoutChanged()), this, SLOT(checkArrivals()));
+    connect(m_model, SIGNAL(rowsInserted(QModelIndex,int,int)), this, SLOT(checkArrivals()));
 }
 
 void MailboxFinder::addMailbox(const QString &mailbox)
diff --git a/src/Imap/Model/Model.cpp b/src/Imap/Model/Model.cpp
index 13bfad83..2bf8f300 100644
--- a/src/Imap/Model/Model.cpp
+++ b/src/Imap/Model/Model.cpp
@@ -122,7 +122,7 @@ Model::Model(QObject *parent, AbstractCache *cache, SocketFactoryPtr socketFacto
 #ifdef DEBUG_PERIODICALLY_DUMP_TASKS
     QTimer *periodicTaskDumper = new QTimer(this);
     periodicTaskDumper->setInterval(1000);
-    connect(periodicTaskDumper, &QTimer::timeout, this, &Model::slotTasksChanged);
+    connect(periodicTaskDumper, SIGNAL(timeout()), this, SLOT(slotTasksChanged()));
     periodicTaskDumper->start();
 #endif
 
@@ -131,7 +131,7 @@ Model::Model(QObject *parent, AbstractCache *cache, SocketFactoryPtr socketFacto
     m_periodicMailboxNumbersRefresh = new QTimer(this);
     // polling every five minutes
     m_periodicMailboxNumbersRefresh->setInterval(5 * 60 * 1000);
-    connect(m_periodicMailboxNumbersRefresh, &QTimer::timeout, this, &Model::invalidateAllMessageCounts);
+    connect(m_periodicMailboxNumbersRefresh, SIGNAL(timeout()), this, SLOT(invalidateAllMessageCounts()));
 }
 
 Model::~Model()
diff --git a/src/Imap/Model/Model.h b/src/Imap/Model/Model.h
index f219ae90..a08359f9 100644
--- a/src/Imap/Model/Model.h
+++ b/src/Imap/Model/Model.h
@@ -310,6 +310,7 @@ public slots:
 
     void invalidateAllMessageCounts();
 
+public:
     QString imapAuthError() const;
 
 private slots:
diff --git a/src/Imap/Model/ModelWatcher.cpp b/src/Imap/Model/ModelWatcher.cpp
index 72cb53db..26fb8cd9 100644
--- a/src/Imap/Model/ModelWatcher.cpp
+++ b/src/Imap/Model/ModelWatcher.cpp
@@ -30,25 +30,25 @@ namespace Mailbox
 
 void ModelWatcher::setModel(QAbstractItemModel *model)
 {
-    connect(model, &QAbstractItemModel::columnsAboutToBeInserted, this, &ModelWatcher::columnsAboutToBeInserted);
-    connect(model, &QAbstractItemModel::columnsAboutToBeRemoved, this, &ModelWatcher::columnsAboutToBeRemoved);
-    connect(model, &QAbstractItemModel::rowsAboutToBeInserted, this, &ModelWatcher::rowsAboutToBeInserted);
-    connect(model, &QAbstractItemModel::rowsAboutToBeRemoved, this, &ModelWatcher::rowsAboutToBeRemoved);
-    connect(model, &QAbstractItemModel::columnsInserted, this, &ModelWatcher::columnsInserted);
-    connect(model, &QAbstractItemModel::columnsRemoved, this, &ModelWatcher::columnsRemoved);
-    connect(model, &QAbstractItemModel::rowsInserted, this, &ModelWatcher::rowsInserted);
-    connect(model, &QAbstractItemModel::rowsRemoved, this, &ModelWatcher::rowsRemoved);
-    connect(model, &QAbstractItemModel::dataChanged, this, &ModelWatcher::dataChanged);
-    connect(model, &QAbstractItemModel::headerDataChanged, this, &ModelWatcher::headerDataChanged);
-
-    connect(model, &QAbstractItemModel::rowsAboutToBeMoved, this, &ModelWatcher::rowsAboutToBeMoved);
-    connect(model, &QAbstractItemModel::rowsMoved, this, &ModelWatcher::rowsMoved);
-    connect(model, &QAbstractItemModel::columnsAboutToBeMoved, this, &ModelWatcher::columnsAboutToBeMoved);
-    connect(model, &QAbstractItemModel::columnsMoved, this, &ModelWatcher::columnsMoved);
-    connect(model, &QAbstractItemModel::layoutAboutToBeChanged, this, &ModelWatcher::layoutAboutToBeChanged);
-    connect(model, &QAbstractItemModel::layoutChanged, this, &ModelWatcher::layoutChanged);
-    connect(model, &QAbstractItemModel::modelAboutToBeReset, this, &ModelWatcher::modelAboutToBeReset);
-    connect(model, &QAbstractItemModel::modelReset, this, &ModelWatcher::modelReset);
+    connect(model, SIGNAL(columnsAboutToBeInserted(QModelIndex,int,int)), this, SLOT(columnsAboutToBeInserted(QModelIndex,int,int)));
+    connect(model, SIGNAL(columnsAboutToBeRemoved(QModelIndex,int,int)), this, SLOT(columnsAboutToBeRemoved(QModelIndex,int,int)));
+    connect(model, SIGNAL(rowsAboutToBeInserted(QModelIndex,int,int)), this, SLOT(rowsAboutToBeInserted(QModelIndex,int,int)));
+    connect(model, SIGNAL(rowsAboutToBeRemoved(QModelIndex,int,int)), this, SLOT(rowsAboutToBeRemoved(QModelIndex,int,int)));
+    connect(model, SIGNAL(columnsInserted(QModelIndex,int,int)), this, SLOT(columnsInserted(QModelIndex,int,int)));
+    connect(model, SIGNAL(columnsRemoved(QModelIndex,int,int)), this, SLOT(columnsRemoved(QModelIndex,int,int)));
+    connect(model, SIGNAL(rowsInserted(QModelIndex,int,int)), this, SLOT(rowsInserted(QModelIndex,int,int)));
+    connect(model, SIGNAL(rowsRemoved(QModelIndex,int,int)), this, SLOT(rowsRemoved(QModelIndex,int,int)));
+    connect(model, SIGNAL(dataChanged(QModelIndex,QModelIndex)), this, SLOT(dataChanged(QModelIndex,QModelIndex)));
+    connect(model, SIGNAL(headerDataChanged(Qt::Orientation,int,int)), this, SLOT(headerDataChanged(Qt::Orientation,int,int)));
+
+    connect(model, SIGNAL(rowsAboutToBeMoved(QModelIndex,int,int,QModelIndex,int)), this, SLOT(rowsAboutToBeMoved(QModelIndex,int,int,QModelIndex,int)));
+    connect(model, SIGNAL(rowsMoved(QModelIndex,int,int,QModelIndex,int)), this, SLOT(rowsMoved(QModelIndex,int,int,QModelIndex,int)));
+    connect(model, SIGNAL(columnsAboutToBeMoved(QModelIndex,int,int,QModelIndex,int)), this, SLOT(columnsAboutToBeMoved(QModelIndex,int,int,QModelIndex,int)));
+    connect(model, SIGNAL(columnsMoved(QModelIndex,int,int,QModelIndex,int)), this, SLOT(columnsMoved(QModelIndex,int,int,QModelIndex,int)));
+    connect(model, SIGNAL(layoutAboutToBeChanged()), this, SLOT(layoutAboutToBeChanged()));
+    connect(model, SIGNAL(layoutChanged()), this, SLOT(layoutChanged()));
+    connect(model, SIGNAL(modelAboutToBeReset()), this, SLOT(modelAboutToBeReset()));
+    connect(model, SIGNAL(modelReset()), this, SLOT(modelReset()));
 }
 
 void ModelWatcher::columnsAboutToBeInserted(const QModelIndex &parent, int start, int end)
diff --git a/src/Imap/Model/NetworkWatcher.cpp b/src/Imap/Model/NetworkWatcher.cpp
index 52ccacce..97007ed5 100644
--- a/src/Imap/Model/NetworkWatcher.cpp
+++ b/src/Imap/Model/NetworkWatcher.cpp
@@ -33,14 +33,14 @@ NetworkWatcher::NetworkWatcher(ImapAccess *parent, Model *model):
 {
     // No assert for m_imapAccess. The test suite doesn't create one.
     Q_ASSERT(m_model);
-    connect(m_model, &Model::networkPolicyChanged, this, &NetworkWatcher::effectiveNetworkPolicyChanged);
-    connect(m_model, &Model::networkError, this, &NetworkWatcher::attemptReconnect);
-    connect(m_model, &Model::connectionStateChanged, this, &NetworkWatcher::handleConnectionStateChanged);
+    connect(m_model, SIGNAL(networkPolicyChanged()), this, SIGNAL(effectiveNetworkPolicyChanged()));
+    connect(m_model, SIGNAL(networkError(QString)), this, SLOT(attemptReconnect()));
+    connect(m_model, SIGNAL(connectionStateChanged(uint,Imap::ConnectionState)), this, SLOT(handleConnectionStateChanged(uint,Imap::ConnectionState)));
 
     m_reconnectTimer = new QTimer(this);
     m_reconnectTimer->setSingleShot(true);
     m_reconnectTimer->setInterval(MIN_RECONNECT_TIMEOUT/2);
-    connect(m_reconnectTimer, &QTimer::timeout, this, &NetworkWatcher::tryReconnect);
+    connect(m_reconnectTimer, SIGNAL(timeout()), this, SLOT(tryReconnect()));
 }
 
 /** @short Start the reconnect attempt cycle */
diff --git a/src/Imap/Model/OneMessageModel.cpp b/src/Imap/Model/OneMessageModel.cpp
index 2905f705..89e4338f 100644
--- a/src/Imap/Model/OneMessageModel.cpp
+++ b/src/Imap/Model/OneMessageModel.cpp
@@ -37,10 +37,10 @@ OneMessageModel::OneMessageModel(Model *model): QObject(model), m_subtree(0)
 {
     m_subtree = new SubtreeModelOfModel(this);
     m_subtree->setSourceModel(model);
-    connect(m_subtree, &QAbstractItemModel::modelReset, this, &OneMessageModel::envelopeChanged);
-    connect(m_subtree, &QAbstractItemModel::dataChanged, this, &OneMessageModel::envelopeChanged);
-    connect(this, &OneMessageModel::envelopeChanged, this, &OneMessageModel::flagsChanged);
-    connect(model, &QAbstractItemModel::dataChanged, this, &OneMessageModel::handleModelDataChanged);
+    connect(m_subtree, SIGNAL(modelReset()), this, SIGNAL(envelopeChanged()));
+    connect(m_subtree, SIGNAL(dataChanged(QModelIndex,QModelIndex)), this, SIGNAL(envelopeChanged()));
+    connect(this, SIGNAL(envelopeChanged()), this, SIGNAL(flagsChanged()));
+    connect(model, SIGNAL(dataChanged(QModelIndex,QModelIndex)), this, SLOT(handleModelDataChanged(QModelIndex,QModelIndex)));
     m_flatteningModel = new KDescendantsProxyModel(this);
     m_flatteningModel->setSourceModel(m_subtree);
     QHash<int, QByteArray> roleNames;
diff --git a/src/Imap/Model/SQLCache.cpp b/src/Imap/Model/SQLCache.cpp
index 1259953a..ca3836f7 100644
--- a/src/Imap/Model/SQLCache.cpp
+++ b/src/Imap/Model/SQLCache.cpp
@@ -80,7 +80,7 @@ void SQLCache::init()
         num = 10000;
     delayedCommit->setInterval(num);
     delayedCommit->setObjectName(QStringLiteral("delayedCommit-%1").arg(objectName()));
-    connect(delayedCommit, &QTimer::timeout, this, &SQLCache::timeToCommit);
+    connect(delayedCommit, SIGNAL(timeout()), this, SLOT(timeToCommit()));
     if (tooMuchTimeWithoutCommit)
         tooMuchTimeWithoutCommit->deleteLater();
     tooMuchTimeWithoutCommit = new QTimer(this);
@@ -89,7 +89,7 @@ void SQLCache::init()
         num = 60000;
     tooMuchTimeWithoutCommit->setInterval(num);
     tooMuchTimeWithoutCommit->setObjectName(QStringLiteral("tooMuchTimeWithoutCommit-%1").arg(objectName()));
-    connect(tooMuchTimeWithoutCommit, &QTimer::timeout, this, &SQLCache::timeToCommit);
+    connect(tooMuchTimeWithoutCommit, SIGNAL(timeout()), this, SLOT(timeToCommit()));
 }
 
 SQLCache::~SQLCache()
diff --git a/src/Imap/Model/SubtreeModel.cpp b/src/Imap/Model/SubtreeModel.cpp
index f057fbbd..10c625a8 100644
--- a/src/Imap/Model/SubtreeModel.cpp
+++ b/src/Imap/Model/SubtreeModel.cpp
@@ -64,10 +64,10 @@ SubtreeModelOfModel::SubtreeModelOfModel(QObject *parent):
 SubtreeModel::SubtreeModel(QObject *parent, SubtreeClassAdaptor *classSpecificAdaptor):
     QAbstractProxyModel(parent), m_classAdaptor(classSpecificAdaptor), m_usingInvalidRoot(false)
 {
-    connect(this, &QAbstractItemModel::layoutChanged, this, &SubtreeModel::validityChanged);
-    connect(this, &QAbstractItemModel::modelReset, this, &SubtreeModel::validityChanged);
-    connect(this, &QAbstractItemModel::rowsInserted, this, &SubtreeModel::validityChanged);
-    connect(this, &QAbstractItemModel::rowsRemoved, this, &SubtreeModel::validityChanged);
+    connect(this, SIGNAL(layoutChanged()), this, SIGNAL(validityChanged()));
+    connect(this, SIGNAL(modelReset()), this, SIGNAL(validityChanged()));
+    connect(this, SIGNAL(rowsInserted(QModelIndex,int,int)), this, SIGNAL(validityChanged()));
+    connect(this, SIGNAL(rowsRemoved(QModelIndex,int,int)), this, SIGNAL(validityChanged()));
 }
 
 SubtreeModel::~SubtreeModel()
@@ -88,15 +88,15 @@ void SubtreeModel::setSourceModel(QAbstractItemModel *sourceModel)
         setRoleNames(sourceModel->roleNames());
 #endif
         // FIXME: will need to be expanded when the source model supports more signals...
-        connect(sourceModel, &QAbstractItemModel::modelAboutToBeReset, this, &SubtreeModel::handleModelAboutToBeReset);
-        connect(sourceModel, &QAbstractItemModel::modelReset, this, &SubtreeModel::handleModelReset);
-        connect(sourceModel, &QAbstractItemModel::layoutAboutToBeChanged, this, &QAbstractItemModel::layoutAboutToBeChanged);
-        connect(sourceModel, &QAbstractItemModel::layoutChanged, this, &QAbstractItemModel::layoutChanged);
-        connect(sourceModel, &QAbstractItemModel::dataChanged, this, &SubtreeModel::handleDataChanged);
-        connect(sourceModel, &QAbstractItemModel::rowsAboutToBeRemoved, this, &SubtreeModel::handleRowsAboutToBeRemoved);
-        connect(sourceModel, &QAbstractItemModel::rowsRemoved, this, &SubtreeModel::handleRowsRemoved);
-        connect(sourceModel, &QAbstractItemModel::rowsAboutToBeInserted, this, &SubtreeModel::handleRowsAboutToBeInserted);
-        connect(sourceModel, &QAbstractItemModel::rowsInserted, this, &SubtreeModel::handleRowsInserted);
+        connect(sourceModel, SIGNAL(modelAboutToBeReset()), this, SLOT(handleModelAboutToBeReset()));
+        connect(sourceModel, SIGNAL(modelReset()), this, SLOT(handleModelReset()));
+        connect(sourceModel, SIGNAL(layoutAboutToBeChanged()), this, SIGNAL(layoutAboutToBeChanged()));
+        connect(sourceModel, SIGNAL(layoutChanged()), this, SIGNAL(layoutChanged()));
+        connect(sourceModel, SIGNAL(dataChanged(QModelIndex,QModelIndex)), this, SLOT(handleDataChanged(QModelIndex,QModelIndex)));
+        connect(sourceModel, SIGNAL(rowsAboutToBeRemoved(QModelIndex,int,int)), this, SLOT(handleRowsAboutToBeRemoved(QModelIndex,int,int)));
+        connect(sourceModel, SIGNAL(rowsRemoved(QModelIndex,int,int)), this, SLOT(handleRowsRemoved(QModelIndex,int,int)));
+        connect(sourceModel, SIGNAL(rowsAboutToBeInserted(QModelIndex,int,int)), this, SLOT(handleRowsAboutToBeInserted(QModelIndex,int,int)));
+        connect(sourceModel, SIGNAL(rowsInserted(QModelIndex,int,int)), this, SLOT(handleRowsInserted(QModelIndex,int,int)));
     }
 }
 
diff --git a/src/Imap/Model/TaskFactory.cpp b/src/Imap/Model/TaskFactory.cpp
index f465f776..42895661 100644
--- a/src/Imap/Model/TaskFactory.cpp
+++ b/src/Imap/Model/TaskFactory.cpp
@@ -216,11 +216,11 @@ Parser *TestingTaskFactory::newParser(Model *model)
 {
     Parser *parser = new Parser(model, model->m_socketFactory->create(), Common::ConnectionId::next());
     ParserState parserState(parser);
-    QObject::connect(parser, &Parser::responseReceived,
-                     model, static_cast<void (Model::*)(Parser *)>(&Model::responseReceived), Qt::QueuedConnection);
-    QObject::connect(parser, &Parser::connectionStateChanged, model, &Model::handleSocketStateChanged);
-    QObject::connect(parser, &Parser::lineReceived, model, &Model::slotParserLineReceived);
-    QObject::connect(parser, &Parser::lineSent, model, &Model::slotParserLineSent);
+    QObject::connect(parser, SIGNAL(responseReceived(Imap::Parser*)),
+                     model, SLOT(responseReceived(Imap::Parser*)), Qt::QueuedConnection);
+    QObject::connect(parser, SIGNAL(connectionStateChanged(Imap::Parser*,Imap::ConnectionState)), model, SLOT(handleSocketStateChanged(Imap::Parser*,Imap::ConnectionState)));
+    QObject::connect(parser, SIGNAL(lineReceived(Imap::Parser*,QByteArray)), model, SLOT(slotParserLineReceived(Imap::Parser*,QByteArray)));
+    QObject::connect(parser, SIGNAL(lineSent(Imap::Parser*,QByteArray)), model, SLOT(slotParserLineSent(Imap::Parser*,QByteArray)));
     model->m_parsers[ parser ] = parserState;
     model->m_taskModel->slotParserCreated(parser);
     return parser;
diff --git a/src/Imap/Model/ThreadingMsgListModel.cpp b/src/Imap/Model/ThreadingMsgListModel.cpp
index 45c9a83d..5e66e1e4 100644
--- a/src/Imap/Model/ThreadingMsgListModel.cpp
+++ b/src/Imap/Model/ThreadingMsgListModel.cpp
@@ -70,7 +70,7 @@ ThreadingMsgListModel::ThreadingMsgListModel(QObject *parent):
     m_delayedPrune = new QTimer(this);
     m_delayedPrune->setSingleShot(true);
     m_delayedPrune->setInterval(0);
-    connect(m_delayedPrune, &QTimer::timeout, this, &ThreadingMsgListModel::delayedPrune);
+    connect(m_delayedPrune, SIGNAL(timeout()), this, SLOT(delayedPrune()));
 }
 
 void ThreadingMsgListModel::setSourceModel(QAbstractItemModel *sourceModel)
@@ -97,14 +97,14 @@ void ThreadingMsgListModel::setSourceModel(QAbstractItemModel *sourceModel)
     QAbstractProxyModel::setSourceModel(msgList);
 
     // FIXME: will need to be expanded when Model supports more signals...
-    connect(sourceModel, &QAbstractItemModel::modelReset, this, &ThreadingMsgListModel::resetMe);
-    connect(sourceModel, &QAbstractItemModel::layoutAboutToBeChanged, this, &QAbstractItemModel::layoutAboutToBeChanged);
-    connect(sourceModel, &QAbstractItemModel::layoutChanged, this, &QAbstractItemModel::layoutChanged);
-    connect(sourceModel, &QAbstractItemModel::dataChanged, this, &ThreadingMsgListModel::handleDataChanged);
-    connect(sourceModel, &QAbstractItemModel::rowsAboutToBeRemoved, this, &ThreadingMsgListModel::handleRowsAboutToBeRemoved);
-    connect(sourceModel, &QAbstractItemModel::rowsRemoved, this, &ThreadingMsgListModel::handleRowsRemoved);
-    connect(sourceModel, &QAbstractItemModel::rowsAboutToBeInserted, this, &ThreadingMsgListModel::handleRowsAboutToBeInserted);
-    connect(sourceModel, &QAbstractItemModel::rowsInserted, this, &ThreadingMsgListModel::handleRowsInserted);
+    connect(sourceModel, SIGNAL(modelReset()), this, SLOT(resetMe()));
+    connect(sourceModel, SIGNAL(layoutAboutToBeChanged()), this, SIGNAL(layoutAboutToBeChanged()));
+    connect(sourceModel, SIGNAL(layoutChanged()), this, SIGNAL(layoutChanged()));
+    connect(sourceModel, SIGNAL(dataChanged(QModelIndex,QModelIndex)), this, SLOT(handleDataChanged(QModelIndex,QModelIndex)));
+    connect(sourceModel, SIGNAL(rowsAboutToBeRemoved(QModelIndex,int,int)), this, SLOT(handleRowsAboutToBeRemoved(QModelIndex,int,int)));
+    connect(sourceModel, SIGNAL(rowsRemoved(QModelIndex,int,int)), this, SLOT(handleRowsRemoved(QModelIndex,int,int)));
+    connect(sourceModel, SIGNAL(rowsAboutToBeInserted(QModelIndex,int,int)), this, SLOT(handleRowsAboutToBeInserted(QModelIndex,int,int)));
+    connect(sourceModel, SIGNAL(rowsInserted(QModelIndex,int,int)), this, SLOT(handleRowsInserted(QModelIndex,int,int)));
     resetMe();
 }
 
@@ -645,14 +645,14 @@ void ThreadingMsgListModel::askForThreading(const uint firstUnknownUid)
                     createIncrementalThreadTask(const_cast<Model *>(realModel), mailboxIndex, requestedAlgorithm,
                                                                     QStringList() << QStringLiteral("INTHREAD") << QString::fromUtf8(requestedAlgorithm) << QStringLiteral("UID") <<
                                                                         QString::fromUtf8(Sequence::startingAt(firstUnknownUid).toByteArray()));
-            connect(threadTask, &ThreadTask::incrementalThreadingAvailable,
-                    this, &ThreadingMsgListModel::slotIncrementalThreadingAvailable);
-            connect(threadTask, &ImapTask::failed, this, &ThreadingMsgListModel::slotIncrementalThreadingFailed);
+            connect(threadTask, SIGNAL(incrementalThreadingAvailable(Responses::ESearch::IncrementalThreadingData_t)),
+                    this, SLOT(slotIncrementalThreadingAvailable(Responses::ESearch::IncrementalThreadingData_t)));
+            connect(threadTask, SIGNAL(failed(QString)), this, SLOT(slotIncrementalThreadingFailed()));
         } else {
             realModel->m_taskFactory->createThreadTask(const_cast<Model *>(realModel), mailboxIndex,
                                                        requestedAlgorithm, QStringList() << QStringLiteral("ALL"));
-            connect(realModel, &Model::threadingAvailable, this, &ThreadingMsgListModel::slotThreadingAvailable);
-            connect(realModel, &Model::threadingFailed, this, &ThreadingMsgListModel::slotThreadingFailed);
+            connect(realModel, SIGNAL(threadingAvailable(QModelIndex,QByteArray,QStringList,QVector<Imap::Responses::ThreadingNode>)), this, SLOT(slotThreadingAvailable(QModelIndex,QByteArray,QStringList,QVector<Imap::Responses::ThreadingNode>)));
+            connect(realModel, SIGNAL(threadingFailed(QModelIndex,QByteArray,QStringList)), this, SLOT(slotThreadingFailed(QModelIndex,QByteArray,QStringList)));
         }
     }
 }
@@ -773,8 +773,8 @@ void ThreadingMsgListModel::slotThreadingFailed(const QModelIndex &mailbox, cons
 
     auto model = qobject_cast<Model*>(sender());
     Q_ASSERT(model);
-    disconnect(model, &Model::threadingAvailable, this, &ThreadingMsgListModel::slotThreadingAvailable);
-    disconnect(model, &Model::threadingFailed, this, &ThreadingMsgListModel::slotThreadingFailed);
+    disconnect(model, SIGNAL(threadingAvailable(QModelIndex,QByteArray,QStringList,QVector<Imap::Responses::ThreadingNode>)), this, SLOT(slotThreadingAvailable(QModelIndex,QByteArray,QStringList,QVector<Imap::Responses::ThreadingNode>)));
+    disconnect(model, SIGNAL(threadingFailed(QModelIndex,QByteArray,QStringList)), this, SLOT(slotThreadingFailed(QModelIndex,QByteArray,QStringList)));
 
     updateNoThreading();
 }
@@ -791,8 +791,8 @@ void ThreadingMsgListModel::slotThreadingAvailable(const QModelIndex &mailbox, c
         return;
 
     Q_ASSERT(model);
-    disconnect(model, &Model::threadingAvailable, this, &ThreadingMsgListModel::slotThreadingAvailable);
-    disconnect(model, &Model::threadingFailed, this, &ThreadingMsgListModel::slotThreadingFailed);
+    disconnect(model, SIGNAL(threadingAvailable(QModelIndex,QByteArray,QStringList,QVector<Imap::Responses::ThreadingNode>)), this, SLOT(slotThreadingAvailable(QModelIndex,QByteArray,QStringList,QVector<Imap::Responses::ThreadingNode>)));
+    disconnect(model, SIGNAL(threadingFailed(QModelIndex,QByteArray,QStringList)), this, SLOT(slotThreadingFailed(QModelIndex,QByteArray,QStringList)));
 
     model->cache()->setMessageThreading(mailbox.data(RoleMailboxName).toString(), mapping);
 
@@ -805,9 +805,9 @@ void ThreadingMsgListModel::slotThreadingAvailable(const QModelIndex &mailbox, c
 void ThreadingMsgListModel::slotSortingAvailable(const Imap::Uids &uids)
 {
     if (!m_sortTask->isPersistent()) {
-        disconnect(m_sortTask.data(), &SortTask::sortingAvailable, this, &ThreadingMsgListModel::slotSortingAvailable);
-        disconnect(m_sortTask.data(), &SortTask::sortingFailed, this, &ThreadingMsgListModel::slotSortingFailed);
-        disconnect(m_sortTask.data(), &SortTask::incrementalSortUpdate, this, &ThreadingMsgListModel::slotSortingIncrementalUpdate);
+        disconnect(m_sortTask.data(), SIGNAL(sortingAvailable(Imap::Uids)), this, SLOT(slotSortingAvailable(Imap::Uids)));
+        disconnect(m_sortTask.data(), SIGNAL(sortingFailed()), this, SLOT(slotSortingFailed()));
+        disconnect(m_sortTask.data(), SIGNAL(incrementalSortUpdate(Imap::Responses::ESearch::IncrementalContextData_t)), this, SLOT(slotSortingIncrementalUpdate(Imap::Responses::ESearch::IncrementalContextData_t)));
 
         m_sortTask = 0;
     }
@@ -820,9 +820,9 @@ void ThreadingMsgListModel::slotSortingAvailable(const Imap::Uids &uids)
 
 void ThreadingMsgListModel::slotSortingFailed()
 {
-    disconnect(m_sortTask.data(), &SortTask::sortingAvailable, this, &ThreadingMsgListModel::slotSortingAvailable);
-    disconnect(m_sortTask.data(), &SortTask::sortingFailed, this, &ThreadingMsgListModel::slotSortingFailed);
-    disconnect(m_sortTask.data(), &SortTask::incrementalSortUpdate, this, &ThreadingMsgListModel::slotSortingIncrementalUpdate);
+    disconnect(m_sortTask.data(), SIGNAL(sortingAvailable(Imap::Uids)), this, SLOT(slotSortingAvailable(Imap::Uids)));
+    disconnect(m_sortTask.data(), SIGNAL(sortingFailed()), this, SLOT(slotSortingFailed()));
+    disconnect(m_sortTask.data(), SIGNAL(incrementalSortUpdate(Imap::Responses::ESearch::IncrementalContextData_t)), this, SLOT(slotSortingIncrementalUpdate(Imap::Responses::ESearch::IncrementalContextData_t)));
 
     m_sortTask = 0;
     m_sortReverse = false;
@@ -1333,9 +1333,9 @@ bool ThreadingMsgListModel::searchSortPreferenceImplementation(const QStringList
             // We have to update our search conditions
             m_sortTask = realModel->m_taskFactory->createSortTask(const_cast<Model *>(realModel), mailboxIndex, searchConditions,
                                                                   QStringList());
-            connect(m_sortTask.data(), &SortTask::sortingAvailable, this, &ThreadingMsgListModel::slotSortingAvailable);
-            connect(m_sortTask.data(), &SortTask::sortingFailed, this, &ThreadingMsgListModel::slotSortingFailed);
-            connect(m_sortTask.data(), &SortTask::incrementalSortUpdate, this, &ThreadingMsgListModel::slotSortingIncrementalUpdate);
+            connect(m_sortTask.data(), SIGNAL(sortingAvailable(Imap::Uids)), this, SLOT(slotSortingAvailable(Imap::Uids)));
+            connect(m_sortTask.data(), SIGNAL(sortingFailed()), this, SLOT(slotSortingFailed()));
+            connect(m_sortTask.data(), SIGNAL(incrementalSortUpdate(Imap::Responses::ESearch::IncrementalContextData_t)), this, SLOT(slotSortingIncrementalUpdate(Imap::Responses::ESearch::IncrementalContextData_t)));
             m_currentSearchConditions = searchConditions;
             m_searchValidity = RESULT_ASKED;
         } else {
@@ -1367,9 +1367,9 @@ bool ThreadingMsgListModel::searchSortPreferenceImplementation(const QStringList
             m_sortTask->cancelSortingUpdates();
 
         m_sortTask = realModel->m_taskFactory->createSortTask(const_cast<Model *>(realModel), mailboxIndex, searchConditions, sortOptions);
-        connect(m_sortTask.data(), &SortTask::sortingAvailable, this, &ThreadingMsgListModel::slotSortingAvailable);
-        connect(m_sortTask.data(), &SortTask::sortingFailed, this, &ThreadingMsgListModel::slotSortingFailed);
-        connect(m_sortTask.data(), &SortTask::incrementalSortUpdate, this, &ThreadingMsgListModel::slotSortingIncrementalUpdate);
+        connect(m_sortTask.data(), SIGNAL(sortingAvailable(Imap::Uids)), this, SLOT(slotSortingAvailable(Imap::Uids)));
+        connect(m_sortTask.data(), SIGNAL(sortingFailed()), this, SLOT(slotSortingFailed()));
+        connect(m_sortTask.data(), SIGNAL(incrementalSortUpdate(Imap::Responses::ESearch::IncrementalContextData_t)), this, SLOT(slotSortingIncrementalUpdate(Imap::Responses::ESearch::IncrementalContextData_t)));
         m_searchValidity = RESULT_ASKED;
     }
 
diff --git a/src/Imap/Network/FileDownloadManager.cpp b/src/Imap/Network/FileDownloadManager.cpp
index 16b6c8df..5b6b10b2 100644
--- a/src/Imap/Network/FileDownloadManager.cpp
+++ b/src/Imap/Network/FileDownloadManager.cpp
@@ -106,10 +106,9 @@ void FileDownloadManager::downloadPart()
     url.setPath(partIndex.data(Imap::Mailbox::RolePartPathToPart).toString());
     request.setUrl(url);
     reply = manager->get(request);
-    connect(reply, &QNetworkReply::finished, this, &FileDownloadManager::onPartDataTransfered);
-    connect(reply, static_cast<void (QNetworkReply::*)(QNetworkReply::NetworkError)>(&QNetworkReply::error),
-            this, &FileDownloadManager::onReplyTransferError);
-    connect(manager, &QNetworkAccessManager::finished, this, &FileDownloadManager::deleteReply);
+    connect(reply, SIGNAL(finished()), this, SLOT(onPartDataTransfered()));
+    connect(reply, SIGNAL(error(QNetworkReply::NetworkError)), this, SLOT(onReplyTransferError()));
+    connect(manager, SIGNAL(finished(QNetworkReply*)), this, SLOT(deleteReply(QNetworkReply*)));
 }
 
 void FileDownloadManager::downloadMessage()
@@ -130,10 +129,10 @@ void FileDownloadManager::downloadMessage()
 
     saving.setFileName(saveFileName);
     saved = false;
-    connect(m_combiner.data(), &Mailbox::FullMessageCombiner::completed, this, &FileDownloadManager::onMessageDataTransferred);
-    connect(m_combiner.data(), &Mailbox::FullMessageCombiner::failed, this, &FileDownloadManager::onCombinerTransferError);
-    connect(m_combiner.data(), &Mailbox::FullMessageCombiner::completed, m_combiner.data(), &QObject::deleteLater, Qt::QueuedConnection);
-    connect(m_combiner.data(), &Mailbox::FullMessageCombiner::failed, m_combiner.data(), &QObject::deleteLater, Qt::QueuedConnection);
+    connect(m_combiner.data(), SIGNAL(completed()), this, SLOT(onMessageDataTransferred()));
+    connect(m_combiner.data(), SIGNAL(failed(QString)), this, SLOT(onCombinerTransferError(QString)));
+    connect(m_combiner.data(), SIGNAL(completed()), m_combiner.data(), SLOT(deleteLater()), Qt::QueuedConnection);
+    connect(m_combiner.data(), SIGNAL(failed(QString)), m_combiner.data(), SLOT(deleteLater()), Qt::QueuedConnection);
     m_combiner->load();
 }
 
diff --git a/src/Imap/Network/MsgPartNetworkReply.cpp b/src/Imap/Network/MsgPartNetworkReply.cpp
index 247faa56..ef06fa22 100644
--- a/src/Imap/Network/MsgPartNetworkReply.cpp
+++ b/src/Imap/Network/MsgPartNetworkReply.cpp
@@ -47,7 +47,7 @@ MsgPartNetworkReply::MsgPartNetworkReply(MsgPartNetAccessManager *parent, const
     setOpenMode(QIODevice::ReadOnly | QIODevice::Unbuffered);
     Q_ASSERT(part.isValid());
 
-    connect(part.model(), &QAbstractItemModel::dataChanged, this, &MsgPartNetworkReply::slotModelDataChanged);
+    connect(part.model(), SIGNAL(dataChanged(QModelIndex,QModelIndex)), this, SLOT(slotModelDataChanged(QModelIndex,QModelIndex)));
 
     // We have to ask for contents before we check whether it's already fetched
     part.data(Imap::Mailbox::RolePartData);
diff --git a/src/Imap/Network/QQuickNetworkReplyWrapper.cpp b/src/Imap/Network/QQuickNetworkReplyWrapper.cpp
index a087fc0a..34b6ae75 100644
--- a/src/Imap/Network/QQuickNetworkReplyWrapper.cpp
+++ b/src/Imap/Network/QQuickNetworkReplyWrapper.cpp
@@ -31,7 +31,7 @@ QQuickNetworkReplyWrapper::QQuickNetworkReplyWrapper(QObject *qQuickReply, QNetw
     Q_ASSERT(qQuickReply);
     Q_ASSERT(qNetworkReply);
     Q_ASSERT(qQuickReply->metaObject()->className() == QByteArray("QQuickNetworkReply"));
-    connect(qNetworkReply, &QNetworkReply::finished, this, &QQuickNetworkReplyWrapper::dataRecieved);
+    connect(qNetworkReply, SIGNAL(finished()), this, SLOT(dataRecieved()));
 }
 
 void QQuickNetworkReplyWrapper::dataRecieved()
diff --git a/src/Imap/Parser/Parser.cpp b/src/Imap/Parser/Parser.cpp
index f0ee8532..a666a7e7 100644
--- a/src/Imap/Parser/Parser.cpp
+++ b/src/Imap/Parser/Parser.cpp
@@ -92,10 +92,10 @@ Parser::Parser(QObject *parent, Streams::Socket *socket, const uint myId):
     waitingForConnection(true), waitingForEncryption(socket->isConnectingEncryptedSinceStart()), waitingForSslPolicy(false),
     m_expectsInitialGreeting(true), readingMode(ReadingLine), oldLiteralPosition(0), m_parserId(myId)
 {
-    connect(socket, &Streams::Socket::disconnected, this, &Parser::handleDisconnected);
-    connect(socket, &Streams::Socket::readyRead, this, &Parser::handleReadyRead);
-    connect(socket, &Streams::Socket::stateChanged, this, &Parser::slotSocketStateChanged);
-    connect(socket, &Streams::Socket::encrypted, this, &Parser::handleSocketEncrypted);
+    connect(socket, SIGNAL(disconnected(QString)), this, SLOT(handleDisconnected(QString)));
+    connect(socket, SIGNAL(readyRead()), this, SLOT(handleReadyRead()));
+    connect(socket, SIGNAL(stateChanged(Imap::ConnectionState,QString)), this, SLOT(slotSocketStateChanged(Imap::ConnectionState,QString)));
+    connect(socket, SIGNAL(encrypted()), this, SLOT(handleSocketEncrypted()));
 }
 
 CommandHandle Parser::noop()
diff --git a/src/Imap/Tasks/FetchMsgPartTask.cpp b/src/Imap/Tasks/FetchMsgPartTask.cpp
index c7a49437..b1cd36be 100644
--- a/src/Imap/Tasks/FetchMsgPartTask.cpp
+++ b/src/Imap/Tasks/FetchMsgPartTask.cpp
@@ -38,7 +38,7 @@ FetchMsgPartTask::FetchMsgPartTask(Model *model, const QModelIndex &mailbox, con
     Q_ASSERT(!uids.isEmpty());
     conn = model->findTaskResponsibleFor(mailboxIndex);
     conn->addDependentTask(this);
-    connect(this, &ImapTask::failed, this, &FetchMsgPartTask::markPendingItemsUnavailable);
+    connect(this, SIGNAL(failed(QString)), this, SLOT(markPendingItemsUnavailable()));
 }
 
 void FetchMsgPartTask::perform()
diff --git a/src/Imap/Tasks/IdleLauncher.cpp b/src/Imap/Tasks/IdleLauncher.cpp
index 8f7e7b84..4382bf0d 100644
--- a/src/Imap/Tasks/IdleLauncher.cpp
+++ b/src/Imap/Tasks/IdleLauncher.cpp
@@ -45,7 +45,7 @@ IdleLauncher::IdleLauncher(KeepMailboxOpenTask *parent):
     if (! ok)
         timeout = 6 * 1000;
     delayedEnter->setInterval(timeout);
-    connect(delayedEnter, &QTimer::timeout, this, &IdleLauncher::slotEnterIdleNow);
+    connect(delayedEnter, SIGNAL(timeout()), this, SLOT(slotEnterIdleNow()));
     renewal = new QTimer(this);
     renewal->setObjectName(QStringLiteral("%1-IdleLauncher-renewal").arg(task->objectName()));
     renewal->setSingleShot(true);
@@ -53,7 +53,7 @@ IdleLauncher::IdleLauncher(KeepMailboxOpenTask *parent):
     if (! ok)
         timeout = 1000 * 29 * 60; // 29 minutes -- that's the longest allowed time to IDLE
     renewal->setInterval(timeout);
-    connect(renewal, &QTimer::timeout, this, &IdleLauncher::slotTerminateLongIdle);
+    connect(renewal, SIGNAL(timeout()), this, SLOT(slotTerminateLongIdle()));
 }
 
 void IdleLauncher::slotEnterIdleNow()
diff --git a/src/Imap/Tasks/ImapTask.cpp b/src/Imap/Tasks/ImapTask.cpp
index 65c6c6c3..75cb094e 100644
--- a/src/Imap/Tasks/ImapTask.cpp
+++ b/src/Imap/Tasks/ImapTask.cpp
@@ -34,7 +34,7 @@ namespace Mailbox
 ImapTask::ImapTask(Model *model) :
     QObject(model), parser(0), parentTask(0), model(model), _finished(false), _dead(false), _aborted(false)
 {
-    connect(this, &QObject::destroyed, model, &Model::slotTaskDying);
+    connect(this, SIGNAL(destroyed(QObject*)), model, SLOT(slotTaskDying(QObject*)));
     CHECK_TASK_TREE;
 }
 
@@ -94,7 +94,7 @@ void ImapTask::markAsActiveTask(const TaskActivatingPosition place)
 
     if (model->accessParser(parser).maintainingTask && model->accessParser(parser).maintainingTask != this) {
         // Got to inform the currently responsible maintaining task about our demise
-        connect(this, &QObject::destroyed, model->accessParser(parser).maintainingTask.data(), &KeepMailboxOpenTask::slotTaskDeleted);
+        connect(this, SIGNAL(destroyed(QObject*)), model->accessParser(parser).maintainingTask.data(), SLOT(slotTaskDeleted(QObject*)));
     }
 
     log(QStringLiteral("Activated"));
diff --git a/src/Imap/Tasks/KeepMailboxOpenTask.cpp b/src/Imap/Tasks/KeepMailboxOpenTask.cpp
index 6d45822c..a0788494 100644
--- a/src/Imap/Tasks/KeepMailboxOpenTask.cpp
+++ b/src/Imap/Tasks/KeepMailboxOpenTask.cpp
@@ -94,7 +94,7 @@ KeepMailboxOpenTask::KeepMailboxOpenTask(Model *model, const QModelIndex &mailbo
 
         // We shall catch destruction of any preexisting tasks so that we can properly launch IDLE etc in response to their termination
         Q_FOREACH(ImapTask *task, model->accessParser(parser).activeTasks) {
-            connect(task, &QObject::destroyed, this, &KeepMailboxOpenTask::slotTaskDeleted);
+            connect(task, SIGNAL(destroyed(QObject*)), this, SLOT(slotTaskDeleted(QObject*)));
         }
     } else {
         ImapTask *conn = 0;
@@ -114,7 +114,7 @@ KeepMailboxOpenTask::KeepMailboxOpenTask(Model *model, const QModelIndex &mailbo
 
     // Setup the timer for NOOPing. It won't get started at this time, though.
     noopTimer = new QTimer(this);
-    connect(noopTimer, &QTimer::timeout, this, &KeepMailboxOpenTask::slotPerformNoop);
+    connect(noopTimer, SIGNAL(timeout()), this, SLOT(slotPerformNoop()));
     bool ok;
     int timeout = model->property("trojita-imap-noop-period").toUInt(&ok);
     if (! ok)
@@ -123,7 +123,7 @@ KeepMailboxOpenTask::KeepMailboxOpenTask(Model *model, const QModelIndex &mailbo
     noopTimer->setSingleShot(true);
 
     fetchPartTimer = new QTimer(this);
-    connect(fetchPartTimer, &QTimer::timeout, this, &KeepMailboxOpenTask::slotFetchRequestedParts);
+    connect(fetchPartTimer, SIGNAL(timeout()), this, SLOT(slotFetchRequestedParts()));
     timeout = model->property("trojita-imap-delayed-fetch-part").toUInt(&ok);
     if (! ok)
         timeout = 50;
@@ -131,7 +131,7 @@ KeepMailboxOpenTask::KeepMailboxOpenTask(Model *model, const QModelIndex &mailbo
     fetchPartTimer->setSingleShot(true);
 
     fetchEnvelopeTimer = new QTimer(this);
-    connect(fetchEnvelopeTimer, &QTimer::timeout, this, &KeepMailboxOpenTask::slotFetchRequestedEnvelopes);
+    connect(fetchEnvelopeTimer, SIGNAL(timeout()), this, SLOT(slotFetchRequestedEnvelopes()));
     fetchEnvelopeTimer->setInterval(0); // message metadata is pretty important, hence an immediate fetch
     fetchEnvelopeTimer->setSingleShot(true);
 
@@ -164,7 +164,7 @@ KeepMailboxOpenTask::KeepMailboxOpenTask(Model *model, const QModelIndex &mailbo
     // arrive, the rate of saving is only moderated based on the number of reponses which were already received,
     // but which did not result in state saving yet.
     m_syncingTimer->setInterval(throttlingWindowLength);
-    connect(m_syncingTimer, &QTimer::timeout, this, &KeepMailboxOpenTask::syncingTimeout);
+    connect(m_syncingTimer, SIGNAL(timeout()), this, SLOT(syncingTimeout()));
 }
 
 void KeepMailboxOpenTask::slotPerformConnection()
@@ -178,7 +178,7 @@ void KeepMailboxOpenTask::slotPerformConnection()
         return;
     }
 
-    connect(synchronizeConn, &QObject::destroyed, this, &KeepMailboxOpenTask::slotTaskDeleted);
+    connect(synchronizeConn, SIGNAL(destroyed(QObject*)), this, SLOT(slotTaskDeleted(QObject*)));
     synchronizeConn->perform();
 }
 
@@ -230,13 +230,13 @@ void KeepMailboxOpenTask::addDependentTask(ImapTask *task)
 
         m_deleteCurrentMailboxTask = deleteTask;
         shouldExit = true;
-        connect(task, &QObject::destroyed, this, &KeepMailboxOpenTask::slotTaskDeleted);
+        connect(task, SIGNAL(destroyed(QObject*)), this, SLOT(slotTaskDeleted(QObject*)));
         ImapTask::addDependentTask(task);
         QTimer::singleShot(0, this, SLOT(slotActivateTasks()));
 
     } else {
         // This branch calls the inherited ImapTask::addDependentTask()
-        connect(task, &QObject::destroyed, this, &KeepMailboxOpenTask::slotTaskDeleted);
+        connect(task, SIGNAL(destroyed(QObject*)), this, SLOT(slotTaskDeleted(QObject*)));
         ImapTask::addDependentTask(task);
         if (task->needsMailbox()) {
             // it's a task which is tied to a particular mailbox
@@ -342,7 +342,7 @@ void KeepMailboxOpenTask::terminate()
         first->keepTaskChild->waitingObtainTasks = waitingObtainTasks + first->keepTaskChild->waitingObtainTasks;
         model->accessParser(parser).maintainingTask = first->keepTaskChild;
         // make sure that if the SELECT dies uncleanly, such as with a missing [CLOSED], we get killed as well
-        connect(first->keepTaskChild, &ImapTask::failed, this, &KeepMailboxOpenTask::finalizeTermination);
+        connect(first->keepTaskChild, SIGNAL(failed(QString)), this, SLOT(finalizeTermination()));
         first->keepTaskChild->slotPerformConnection();
     } else {
         Q_ASSERT(dependentTasks.isEmpty());
@@ -961,7 +961,7 @@ bool KeepMailboxOpenTask::dieIfInvalidMailbox()
     // See ObtainSynchronizedMailboxTask::dieIfInvalidMailbox() for details
     if (!unSelectTask && isRunning == Running::RUNNING) {
         unSelectTask = model->m_taskFactory->createUnSelectTask(model, this);
-        connect(unSelectTask, &ImapTask::completed, this, &KeepMailboxOpenTask::slotUnselected);
+        connect(unSelectTask, SIGNAL(completed(Imap::Mailbox::ImapTask*)), this, SLOT(slotUnselected()));
         unSelectTask->perform();
     }
 
diff --git a/src/Imap/Tasks/ObtainSynchronizedMailboxTask.cpp b/src/Imap/Tasks/ObtainSynchronizedMailboxTask.cpp
index 9d6f8279..d63432e8 100644
--- a/src/Imap/Tasks/ObtainSynchronizedMailboxTask.cpp
+++ b/src/Imap/Tasks/ObtainSynchronizedMailboxTask.cpp
@@ -51,7 +51,7 @@ ObtainSynchronizedMailboxTask::ObtainSynchronizedMailboxTask(Model *model, const
     CHECK_TASK_TREE
     addDependentTask(keepTaskChild);
     CHECK_TASK_TREE
-    connect(this, &ImapTask::failed, this, &ObtainSynchronizedMailboxTask::signalSyncFailure);
+    connect(this, SIGNAL(failed(QString)), this, SLOT(signalSyncFailure(QString)));
 }
 
 void ObtainSynchronizedMailboxTask::addDependentTask(ImapTask *task)
@@ -1189,7 +1189,7 @@ bool ObtainSynchronizedMailboxTask::dieIfInvalidMailbox()
 
     if (!unSelectTask) {
         unSelectTask = model->m_taskFactory->createUnSelectTask(model, this);
-        connect(unSelectTask, &ImapTask::completed, this, &ObtainSynchronizedMailboxTask::slotUnSelectCompleted);
+        connect(unSelectTask, SIGNAL(completed(Imap::Mailbox::ImapTask*)), this, SLOT(slotUnSelectCompleted()));
         unSelectTask->perform();
     }
 
diff --git a/src/Imap/Tasks/OpenConnectionTask.cpp b/src/Imap/Tasks/OpenConnectionTask.cpp
index de676ee9..37f6f5cd 100644
--- a/src/Imap/Tasks/OpenConnectionTask.cpp
+++ b/src/Imap/Tasks/OpenConnectionTask.cpp
@@ -43,10 +43,10 @@ OpenConnectionTask::OpenConnectionTask(Model *model) :
     Q_ASSERT(model->networkPolicy() != NETWORK_OFFLINE);
     parser = new Parser(model, model->m_socketFactory->create(), Common::ConnectionId::next());
     ParserState parserState(parser);
-    connect(parser, &Parser::responseReceived, model, static_cast<void (Model::*)(Parser*)>(&Model::responseReceived), Qt::QueuedConnection);
-    connect(parser, &Parser::connectionStateChanged, model, &Model::handleSocketStateChanged);
-    connect(parser, &Parser::lineReceived, model, &Model::slotParserLineReceived);
-    connect(parser, &Parser::lineSent, model, &Model::slotParserLineSent);
+    connect(parser, SIGNAL(responseReceived(Imap::Parser*)), model, SLOT(responseReceived(Imap::Parser*)), Qt::QueuedConnection);
+    connect(parser, SIGNAL(connectionStateChanged(Imap::Parser*,Imap::ConnectionState)), model, SLOT(handleSocketStateChanged(Imap::Parser*,Imap::ConnectionState)));
+    connect(parser, SIGNAL(lineReceived(Imap::Parser*,QByteArray)), model, SLOT(slotParserLineReceived(Imap::Parser*,QByteArray)));
+    connect(parser, SIGNAL(lineSent(Imap::Parser*,QByteArray)), model, SLOT(slotParserLineSent(Imap::Parser*,QByteArray)));
     model->m_parsers[ parser ] = parserState;
     model->m_taskModel->slotParserCreated(parser);
     markAsActiveTask();
diff --git a/src/Imap/Tasks/UnSelectTask.cpp b/src/Imap/Tasks/UnSelectTask.cpp
index e8c36398..639c6d5f 100644
--- a/src/Imap/Tasks/UnSelectTask.cpp
+++ b/src/Imap/Tasks/UnSelectTask.cpp
@@ -38,7 +38,7 @@ UnSelectTask::UnSelectTask(Model *model, ImapTask *parentTask) :
     conn = parentTask;
     parser = conn->parser;
     Q_ASSERT(parser);
-    connect(this, &ImapTask::completed, this, &UnSelectTask::resetConnectionState);
+    connect(this, SIGNAL(completed(Imap::Mailbox::ImapTask*)), this, SLOT(resetConnectionState()));
 }
 
 void UnSelectTask::perform()
diff --git a/tests/Composer/test_Html_formatting.cpp b/tests/Composer/test_Html_formatting.cpp
index 960fefb7..2847b7cc 100644
--- a/tests/Composer/test_Html_formatting.cpp
+++ b/tests/Composer/test_Html_formatting.cpp
@@ -230,7 +230,7 @@ WebRenderingTester::WebRenderingTester()
 {
     m_web = new QWebView(0);
     m_loop = new QEventLoop(this);
-    connect(m_web, &QWebView::loadFinished, m_loop, &QEventLoop::quit);
+    connect(m_web, SIGNAL(loadFinished(bool)), m_loop, SLOT(quit()));
 }
 
 WebRenderingTester::~WebRenderingTester()
diff --git a/tests/Imap/test_Imap_SelectedMailboxUpdates.cpp b/tests/Imap/test_Imap_SelectedMailboxUpdates.cpp
index d82192f1..37acd99d 100644
--- a/tests/Imap/test_Imap_SelectedMailboxUpdates.cpp
+++ b/tests/Imap/test_Imap_SelectedMailboxUpdates.cpp
@@ -987,7 +987,7 @@ void ImapModelSelectedMailboxUpdatesTest::testMarkAllConcurrentArrival()
     cEmpty();
 
     QSignalSpy changedSpy(model, SIGNAL(dataChanged(QModelIndex,QModelIndex)));
-    connect(model, &QAbstractItemModel::dataChanged, this, &ImapModelSelectedMailboxUpdatesTest::helperDataChangedUidNonZero);
+    connect(model, SIGNAL(dataChanged(QModelIndex,QModelIndex)), this, SLOT(helperDataChangedUidNonZero(QModelIndex,QModelIndex)));
     QSignalSpy messageCountChangedSpy(model, SIGNAL(messageCountPossiblyChanged(QModelIndex)));
     model->markMailboxAsRead(idxA);
     cClient(t.mk("STORE 1:* +FLAGS.SILENT \\Seen\r\n"));
diff --git a/tests/Imap/test_Imap_Tasks_ListChildMailboxes.cpp b/tests/Imap/test_Imap_Tasks_ListChildMailboxes.cpp
index b81c93ff..de3e8770 100644
--- a/tests/Imap/test_Imap_Tasks_ListChildMailboxes.cpp
+++ b/tests/Imap/test_Imap_Tasks_ListChildMailboxes.cpp
@@ -35,6 +35,7 @@
 void ImapModelListChildMailboxesTest::init()
 {
     m_fakeListCommand = false;
+    m_autoExpandingFinder = nullptr;
     LibMailboxSync::init();
 
     LibMailboxSync::setModelNetworkPolicy(model, Imap::Mailbox::NETWORK_ONLINE);
@@ -239,17 +240,12 @@ void ImapModelListChildMailboxesTest::testAutoExpanding()
 {
     Imap::Mailbox::MailboxModel mm(nullptr, model);
     Imap::Mailbox::MailboxFinder finder(nullptr, &mm);
+    m_autoExpandingFinder = &finder;
 
-    auto resetWatchedMailboxes = [&finder]() {
-        finder.addMailbox(QStringLiteral("a"));
-        finder.addMailbox(QStringLiteral("a.A"));
-        finder.addMailbox(QStringLiteral("a.A.1"));
-    };
-
-    resetWatchedMailboxes();
-    connect(&mm, &QAbstractItemModel::layoutChanged, this, resetWatchedMailboxes);
-    connect(&mm, &QAbstractItemModel::rowsRemoved, this, resetWatchedMailboxes);
-    connect(&mm, &QAbstractItemModel::modelReset, this, resetWatchedMailboxes);
+    helperResetAutoExpandingMailboxes();
+    connect(&mm, SIGNAL(layoutChanged()), this, SLOT(helperResetAutoExpandingMailboxes()));
+    connect(&mm, SIGNAL(rowsRemoved(QModelIndex,int,int)), this, SLOT(helperResetAutoExpandingMailboxes()));
+    connect(&mm, SIGNAL(modelReset()), this, SLOT(helperResetAutoExpandingMailboxes()));
 
     cClient(t.mk("LIST \"\" \"%\"\r\n"));
     cServer("* LIST (\\HasChildren) \".\" a\r\n"
@@ -295,5 +291,17 @@ void ImapModelListChildMailboxesTest::testAutoExpanding()
     cEmpty();
 }
 
+/** @short Helper for testAutoExpanding(), (re-)registers interest in the watched mailboxes */
+void ImapModelListChildMailboxesTest::helperResetAutoExpandingMailboxes()
+{
+    if (!m_autoExpandingFinder) {
+        // hack: don't let QTestLib "run" this method in a standalone manner
+        return;
+    }
+    m_autoExpandingFinder->addMailbox(QStringLiteral("a"));
+    m_autoExpandingFinder->addMailbox(QStringLiteral("a.A"));
+    m_autoExpandingFinder->addMailbox(QStringLiteral("a.A.1"));
+}
+
 
 TROJITA_HEADLESS_TEST( ImapModelListChildMailboxesTest )
diff --git a/tests/Imap/test_Imap_Tasks_ListChildMailboxes.h b/tests/Imap/test_Imap_Tasks_ListChildMailboxes.h
index 71ed1d8f..e57d89e7 100644
--- a/tests/Imap/test_Imap_Tasks_ListChildMailboxes.h
+++ b/tests/Imap/test_Imap_Tasks_ListChildMailboxes.h
@@ -30,6 +30,12 @@
 
 class QSignalSpy;
 
+namespace Imap {
+namespace Mailbox {
+class MailboxFinder;
+}
+}
+
 class ImapModelListChildMailboxesTest : public LibMailboxSync
 {
     Q_OBJECT
@@ -46,6 +52,10 @@ private slots:
     void testFailingList();
 
     void testAutoExpanding();
+
+    void helperResetAutoExpandingMailboxes();
+private:
+    Imap::Mailbox::MailboxFinder *m_autoExpandingFinder;
 };
 
 #endif
diff --git a/tests/Imap/test_Imap_Tasks_OpenConnection.cpp b/tests/Imap/test_Imap_Tasks_OpenConnection.cpp
index 81cef20a..d1a26c45 100644
--- a/tests/Imap/test_Imap_Tasks_OpenConnection.cpp
+++ b/tests/Imap/test_Imap_Tasks_OpenConnection.cpp
@@ -53,8 +53,8 @@ void ImapModelOpenConnectionTest::reinit(const TlsRequired tlsRequired)
     m_initialConnectionState = Imap::CONN_STATE_CONNECTED_PRETLS_PRECAPS;
     LibMailboxSync::init();
     model->setProperty("trojita-imap-id-no-versions", QVariant(true));
-    connect(model, &Imap::Mailbox::Model::authRequested, this, &ImapModelOpenConnectionTest::provideAuthDetails, Qt::QueuedConnection);
-    connect(model, &Imap::Mailbox::Model::needsSslDecision, this, &ImapModelOpenConnectionTest::acceptSsl, Qt::QueuedConnection);
+    connect(model, SIGNAL(authRequested()), this, SLOT(provideAuthDetails()), Qt::QueuedConnection);
+    connect(model, SIGNAL(needsSslDecision(QList<QSslCertificate>,QList<QSslError>)), this, SLOT(acceptSsl(QList<QSslCertificate>,QList<QSslError>)), Qt::QueuedConnection);
     LibMailboxSync::setModelNetworkPolicy(model, Imap::Mailbox::NETWORK_ONLINE);
     QCoreApplication::processEvents();
     task = new Imap::Mailbox::OpenConnectionTask(model);
diff --git a/tests/Imap/test_Imap_Threading.cpp b/tests/Imap/test_Imap_Threading.cpp
index 99b78c29..dc9987c8 100644
--- a/tests/Imap/test_Imap_Threading.cpp
+++ b/tests/Imap/test_Imap_Threading.cpp
@@ -1395,9 +1395,9 @@ void ImapModelThreadingTest::testMultipleExpunges()
     // index was, however, not updated by the delayedPrune method, and therefore it was left dangling. In the GUI, this
     // was apparent because some items were suddenly getting selected after some other messages were removed, and the visual
     // position of the now bogous selection was conspicuously similar to the expunged messages.
-    connect(threadingModel, &QAbstractItemModel::layoutAboutToBeChanged, this, &ImapModelThreadingTest::helper_multipleExpunges);
+    connect(threadingModel, SIGNAL(layoutAboutToBeChanged()), this, SLOT(helper_multipleExpunges()));
     cServer("* 2 EXPUNGE\r\n* 2 EXPUNGE\r\n");
-    disconnect(threadingModel, &QAbstractItemModel::layoutAboutToBeChanged, this, &ImapModelThreadingTest::helper_multipleExpunges);
+    disconnect(threadingModel, SIGNAL(layoutAboutToBeChanged()), this, SLOT(helper_multipleExpunges()));
     QCOMPARE(helper_multipleExpunges_hit, 1);
 
     QCOMPARE(QString::fromUtf8(treeToThreading(QModelIndex())), QString::fromUtf8("(1)(4)"));
diff --git a/tests/Utils/LibMailboxSync.cpp b/tests/Utils/LibMailboxSync.cpp
index d35d4e83..8444651b 100644
--- a/tests/Utils/LibMailboxSync.cpp
+++ b/tests/Utils/LibMailboxSync.cpp
@@ -75,8 +75,8 @@ void LibMailboxSync::setupLogging()
 {
     errorSpy = new QSignalSpy(model, SIGNAL(imapError(QString)));
     netErrorSpy = new QSignalSpy(model, SIGNAL(networkError(QString)));
-    connect(model, &Imap::Mailbox::Model::imapError, this, &LibMailboxSync::modelSignalsError);
-    connect(model, &Imap::Mailbox::Model::logged, this, &LibMailboxSync::modelLogged);
+    connect(model, SIGNAL(imapError(QString)), this, SLOT(modelSignalsError(QString)));
+    connect(model, SIGNAL(logged(uint,Common::LogMessage)), this, SLOT(modelLogged(uint,Common::LogMessage)));
 }
 
 void LibMailboxSync::init()
-- 
2.43.0

